File: src/main/java/com/viartemev/requestmapper/annotations/spring/PutMapping.java
package com.viartemev.requestmapper.annotations.spring;

import com.intellij.psi.PsiAnnotation;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiMethod;
import com.viartemev.requestmapper.RequestMappingItem;

import java.util.Collections;
import java.util.List;

public class PutMapping extends RequestMapping {
    private static final String METHOD = "PUT";

    public PutMapping(PsiAnnotation psiAnnotation, PsiElement psiElement) {
        super(psiAnnotation, psiElement);
    }

    @Override
    public List<RequestMappingItem> values() {
        if (psiElement instanceof PsiMethod) {
            return fetchRequestMappingItem(psiAnnotation, (PsiMethod) psiElement, METHOD);
        }
        return Collections.emptyList();
    }

}

File: src/main/kotlin/com/viartemev/requestmapper/annotations/spring/PutMapping.kt
package com.viartemev.requestmapper.annotations.spring

import com.intellij.psi.PsiAnnotation

class PutMapping(psiAnnotation: PsiAnnotation) : SpringMappingAnnotation(psiAnnotation) {

    override fun extractMethod() = METHOD

    companion object {
        private val METHOD = "PUT"
    }

}
